packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -825,13 +875,9 @@ export const createAssetNodes = ({
     // The content of an entry is guaranteed to be updated if and only if the .sys.updatedAt field changed
     assetNode.internal.contentDigest = assetItem.sys.updatedAt
 
-    // if the node hasn't changed, createNode may return `undefined` instead of a Promise on some versions of Gatsby
-    const maybePromise = createNode(assetNode)
-
-    createNodePromises.push(
-      maybePromise?.then ? maybePromise.then(() => assetNode) : assetNode
-    )
+    assetNodes.push(assetNode)
+    create(assetNode)
   })
 
-  return createNodePromises
+  return createNodesPromise.then(() => assetNodes)
```

Comment:
Can't this be a async function as well? 🤔

Reply:
It can but I wanted to make the relationship between the promise and asset nodes really clear here since the createNodesPromise doesn't return nodes. I'll change it though since I dont' think it's that helpful